tests/client/amo/testApi.js

```diff
@@ -165,86 +138,63 @@ describe('amo.api', () => {
+        .returns(Promise.resolve({ results: [fakeReview] }));
 
-      return getUserReviews({ userId: 123, addonId: fakeReview.addon.id })
+      return getReviews(params)
         .then((reviews) => {
           assert.deepEqual(reviews, [fakeReview]);
+          mockApi.verify();
         });
     });
-  });
 
-  describe('getAddonReviews', () => {
-    it('gets all add-on reviews', () => {
-      const addonSlug = 'ublock';
-      const reviewList = [fakeReview];
-      const response = { results: reviewList };
+    it('requires a user or addon', () => {
       mockApi
         .expects('callApi')
-        .once()
-        .withArgs({
-          method: 'GET',
-          endpoint: `addons/addon/${addonSlug}/reviews`,
-        })
-        .returns(Promise.resolve(response));
+        .returns(Promise.resolve({ results: [fakeReview] }));
 
-      return getAddonReviews({ addonSlug })
-        .then((reviews) => {
-          mockApi.verify();
-          assert.deepEqual(reviews, reviewList);
-        });
-    });
-
-    it('requires a truthy addonSlug', () => {
-      return getAddonReviews()
+      return getReviews()
         .then(() => assert(false, 'Unexpected success'), (error) => {
-          assert.match(error.message, /addonSlug cannot be falsey/);
+          assert.match(error.message, /user or addon must be specified/);
         });
     });
+  });
 
-    it('does not support paging yet', () => {
+  describe('getLatestUserReview', () => {
+    it('returns the lone review result since that is the latest', () => {
       mockApi
         .expects('callApi')
-        .returns(Promise.resolve({
-          results: [],
-          next: '/reviews/next-page/',
-        }));
+        .returns(Promise.resolve({ results: [fakeReview] }));
 
-      // This will log a warning so just make sure it doesn't throw.
-      return getAddonReviews({ addonSlug: 'ublock' });
+      return getLatestUserReview({ user: 123, addon: 321 })
+        .then((review) => {
+          assert.deepEqual(review, fakeReview);
+        });
     });
-  });
 
-  describe('getLatestUserReview', () => {
-    it('allows you to fetch only the latest review', () => {
-      const latestReview = { ...fakeReview, is_latest: true };
+    it('throws an error if multple reviews are received', () => {
       mockApi
         .expects('callApi')
         .returns(Promise.resolve({
+          // In real life, the API should never return multiple reviews like this.
           results: [
             fakeReview,
-            latestReview,
+            { id: 456, ...fakeReview },
           ],
         }));
 
-      return getLatestUserReview({
-        userId: 123,
-        addonId: fakeReview.addon.id,
-      })
-        .then((review) => {
-          assert.deepEqual(review, latestReview);
+      return getLatestUserReview({ user: 123, addon: fakeReview.addon.id })
+        .then(() => assert(false, 'unexpected success'), (error) => {
```

Comment:
We have an `unexpectedSuccess` helper that prevents writing out the `() => assert(false, 'unexpected success')` as many times and I think makes the code cleaner.

Also, why not use `catch` here instead of the second promise argument? (We usually use catch)